src/libraries/Microsoft.PowerFx.Core/Syntax/UserDefinitions.cs

```diff
@@ -155,5 +166,121 @@ private bool CheckReturnType(IdentToken returnType, List<TexlError> errors)
+                    {
+                        errors.Add(new TexlError(formula.Attribute.AttributeOperationToken, DocumentErrorSeverity.Severe, TexlStrings.ErrOnlyPartialAttribute));
+                        continue;
+                    }
+
+                    if (!firstAttribute.SameAttribute(formula.Attribute))
+                    {
+                        errors.Add(new TexlError(formula.Attribute.AttributeOperationToken, DocumentErrorSeverity.Severe, TexlStrings.ErrOperationDoesntMatch));
+                        continue;
+                    }
+
+                    var newName = new IdentToken(name + _renamedFormulaGuid + id, formula.Ident.Span, isNonSourceIdentToken: true);
+                    id++;
+                    updatedGroupFormulas.Add(new NamedFormula(newName, formula.Formula, formula.StartingIndex, formula.Attribute));
+                }
+
+                if (updatedGroupFormulas.Count != nameGroup.Count())
+                {
+                    // Not all matched, don't use renamed formulas
+                    newFormulas.AddRange(nameGroup);
+                    continue;
+                }
+
+                newFormulas.AddRange(updatedGroupFormulas);
+                newFormulas.Add(
+                    new NamedFormula(
+                        new IdentToken(name, firstAttribute.AttributeName.Span, isNonSourceIdentToken: true),
+                        GetPartialCombinedFormula(name, firstAttribute.AttributeOperation, updatedGroupFormulas),
+                        0,
+                        firstAttribute));
+            }
+
+            return new ParseUserDefinitionResult(newFormulas, parsed.UDFs, parsed.DefinedTypes, errors, parsed.Comments);
+        }
+
+        private Formula GetPartialCombinedFormula(string name, PartialAttribute.AttributeOperationKind operationKind, IList<NamedFormula> formulas)
+        {
+            return operationKind switch
+            {
+                PartialAttribute.AttributeOperationKind.PartialAnd => GeneratePartialFunction("And", name, formulas),
+                PartialAttribute.AttributeOperationKind.PartialOr => GeneratePartialFunction("Or", name, formulas),
+                PartialAttribute.AttributeOperationKind.PartialTable => GeneratePartialFunction("TableConcatenate", name, formulas),
+                PartialAttribute.AttributeOperationKind.PartialRecord => GeneratePartialFunction("MergeRecords", name, formulas),
+                _ => throw new InvalidOperationException("Unknown partial op while generating merged NF")
+            };
+        }
+
+        private Formula GeneratePartialFunction(string functionName, string name, IList<NamedFormula> formulas)
+        {
+            // We're going to construct these texlnodes by hand so the spans match up with real code locations
+            var script = $"{functionName}({string.Join(", ", Enumerable.Range(0, formulas.Count).Select(i => name + _renamedFormulaGuid + i))})";
```

Comment:
>,  [](http://example.com/codeflow?start=56&length=2)

Does it work with non-en-us locales, such as fr-FR, where the list separator is `;`?

Reply:
ah, good point. This is just a script, and we're manually constructing the parsetree below, but it should still be the real locale-specific separator.